pygfx/utils/load.py

```diff
@@ -5,72 +5,432 @@
+    remote_ok : bool
+        Whether to allow loading meshes from URLs, by default False.
 
     Returns
     -------
-    mesh : Mesh
-        A pygfx Mesh object. If the file does not contain exactly one mesh, an error is raised.
+    meshes : list
+        A list of loaded meshes.
+
     """
-    meshes = load_meshes(path)
-    if len(meshes) != 1:
-        raise ValueError(
-            f"Found {len(meshes)} meshes instead of 1 in '{path}'. Use `load_meshes()` instead. "
-        )
-    return meshes[0]
+    raise DeprecationWarning("The load_meshes() function is replaced with load_mesh().")
 
 
-def load_meshes(path):
-    """Load meshes from a file.
+def load_mesh(path, remote_ok=False):
+    """Load mesh(es) from a file.
 
     This function requires the trimesh library.
 
     Parameters
     ----------
     path : str
         The location where the mesh or scene is stored.
+        Can be a local file path or a URL.
+    remote_ok : bool
+        Whether to allow loading meshes from URLs, by default False.
 
     Returns
     -------
     meshes : list
-        A list of loaded meshes.
+        A list of pygfx.Meshes.
+
+    See Also
+    --------
+    load_scene
+        Loads the entire scene (lights, graph, etc.) from a file.
 
     """
+    import trimesh  # noqa
+
+    # Trimesh's load() performs a similar check and refers
+    # loading from URLs to load_remote()
+    if "https://" in str(path) or "http://" in str(path):
+        if not remote_ok:
+            raise ValueError(
+                "Loading meshes from URLs is disabled. "
+                "Set remote_ok=True to allow loading from URLs."
+            )
+        scene = trimesh.load_remote(path)
+    else:
+        scene = trimesh.load(path)
+
+    return meshes_from_trimesh(scene, apply_transforms=True)
+
 
+def load_scene(
+    path,
+    flatten=False,
+    meshes=True,
+    materials=True,
+    lights=True,
+    camera=True,
+    remote_ok=False,
+):
+    """Load file into a scene.
+
+    When reading file formats that can contain more than just meshes (e.g. `.glb`) this function will
+    attempt to import the entire scene, including lights, cameras, and materials. If the file format
+    is mesh-only (e.g. `.stl`) we will instead construct a minimal scene.
+
+    This function requires the trimesh library.
+
+    Parameters
+    ----------
+    path : str
+        The filepath. Can be a local file or a URL.
+    flatten : bool
+        If True, will ignore any hierarchical structure in the scene graph and
+        instead import as a flat list of objects.
+    meshes : bool
+        Whether to load meshes.
+    materials : bool
+        Whether to load materials.
+    lights : bool, optional
+        Whether to load lights:
+          - True (default): lights are imported if present but if not, minimal lights are added
+          - False: ignore any lights in the file but add minimal lights to scene
+          - None: no lights are loaded
```

Comment:
If find the "enum"  a bit awkward. Users that do not load the lights, can easily add a minimal light with a on-liner, I think?

Reply:
Yeah, maybe it's a bit over-engineered. These are the use cases I was thinking of:

1. Users that just want a scene that works and looks good, no matter whether it all came from the file itself (`True`).
2. Users that want _exactly_ what's in the file - nothing more, nothing less (currently not covered).
3. Users that do not want to load any lights - e.g. perhaps because the lights in the file are broken or because they want to use their own lights (`None`).
4. Users that want none of the lights in the file but would like _some_ lights so the scene is ready to go (`False`).

I suspect the order of frequency would be 1 >> 2 > 3 > 4. The current behaviour is not actually capturing that. So perhaps the arguments should change to:
- `True` = 1
- `None` = 2
- `False` = 3

That said: I'm also perfectly happy to keep it simple with a just `True` and `False` for `1` and `3`, respectively.

Reply:
Then perhaps this argument is actually an enum, and we should make it an actual enum (allowing the user to select the option using a string).

Reply:
Perhaps something like this then?
```
lights :  "auto" | "file" | "none"
    Whether to import lights. "auto" (default) will always add lights to the scene
    even if the file does not contain any; "file" will import only lights defined in
    the file; "none" will not add any lights to the scene.
```

Reply:
@almarklein please have a look if you like the design introduced with 8a6ec2f better?

Reply:
The new enums look great!